### DISCUSSION TITLE
question on RobotWrapper::frameClassicAcceleration()

### DISCUSSION THREAD

**Discussion by xiaosuyang** [stack-of-tasks/tsid#229] [Q&A] [answered]
Hello.
In the tsid robotwrapper.cpp, function RobotWrapper::frameClassicAcceleration() which is used to calculate the classic accleration of a frame:

![image](https://github.com/stack-of-tasks/tsid/assets/55594017/39869254-47e3-4e63-acc8-10637e69bf5a)

In row 301 and 302, a and v are spatial acc and spatial vel of the frame f, but why they use placement.actInv(data.v[f.parent])(data.v[f.parent] is the vel of the parent joint ) instead of placement.act(). Isn't the frameXJoint.act() used to transform the vector from Joint coordinate to the frame coordinate?

**Comment by andreadelprete** [comment, marked as answer]
Dear @xiaosuyang ,

the reason is that the `f.placement` variable contains the transformation `joint_X_frame`.

**Reply by xiaosuyang** [reply]
Thank you for reply,  it means that f.placement.actinv() transform the vector from joint frame to the index frame ?

**Reply by andreadelprete** [reply]
Exactly!
